source/iot/MqttClient.cpp

```diff
@@ -45,6 +45,33 @@ namespace Aws
             };
         }
 
+        WebsocketConfig::WebsocketConfig(const Crt::String &signingRegion, Crt::Allocator *allocator) noexcept
+            : SigningRegion(signingRegion), ServiceName("iotdevicegateway")
+        {
+            Crt::Auth::CredentialsProviderChainDefaultConfig config;
+            config.Bootstrap = Crt::ApiHandle::GetOrCreateStaticDefaultClientBootstrap();
+
+            CredentialsProvider =
+                Crt::Auth::CredentialsProvider::CreateCredentialsProviderChainDefault(config, allocator);
```

Comment:
No need to copy/paste so much code. You can have this new constructor call into another one like:
```
WebsocketConfig::WebsocketConfig(const Crt::String &signingRegion, Crt::Allocator *allocator) noexcept
    : WebsocketConfig(signingRegion, ApiHandle::GetOrCreateDefaultClientBootstrap(), allocator)
{}
```